examples/tutorials/advanced/Federated Recurrent Neural Network.ipynb

```diff
@@ -0,0 +1,1133 @@
+    "\n",
+    "class RNN(nn.Module):\n",
+    "    def __init__(self, input_size, hidden_size, output_size):\n",
+    "        super(RNN, self).__init__()\n",
+    "        self.hidden_size = hidden_size\n",
+    "        self.i2h = nn.Linear(input_size + hidden_size, hidden_size)\n",
+    "        self.i2o = nn.Linear(input_size + hidden_size, output_size)\n",
+    "        self.softmax = nn.LogSoftmax(dim=1)\n",
+    "\n",
+    "    def forward(self, input, hidden):\n",
+    "        combined = torch.cat((input, hidden), 1)\n",
+    "        hidden = self.i2h(combined)\n",
+    "        output = self.i2o(combined)\n",
+    "        output = self.softmax(output)\n",
+    "        return output, hidden\n",
+    "\n",
+    "    def initHidden(self):\n",
+    "        return torch.zeros(1, self.hidden_size)\n",
+    "\n",
+    "#Let's instantiate the neural network already:\n",
+    "n_hidden = 128\n",
+    "#Instantiate RNN\n",
+    "\n",
+    "device = torch.device(\"cuda\" if args.use_cuda else \"cpu\")\n",
+    "model = RNN(n_letters, n_hidden, n_categories).to(device)\n",
+    "#The final softmax layer will produce a probability for each one of our 18 categories\n",
+    "print(model)    \n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Now let's define our workers. You can either use remote workers or virtual workers\n",
+    "hook = sy.TorchHook(torch)  # <-- NEW: hook PyTorch ie add extra functionalities to support Federated Learning\n",
+    "alice = sy.VirtualWorker(hook, id=\"alice\")  \n",
+    "bob = sy.VirtualWorker(hook, id=\"bob\")  \n",
+    "#charlie = sy.VirtualWorker(hook, id=\"charlie\") \n",
+    "\n",
+    "workers_virtual = [alice, bob]\n",
+    "\n",
+    "#If you have your workers operating remotely, like on Raspberry PIs\n",
+    "#kwargs_websocket_alice = {\"host\": \"ip_alice\", \"hook\": hook}\n",
+    "#alice = WebsocketClientWorker(id=\"alice\", port=8777, **kwargs_websocket_alice)\n",
+    "#kwargs_websocket_bob = {\"host\": \"ip_bob\", \"hook\": hook}\n",
+    "#bob = WebsocketClientWorker(id=\"bob\", port=8778, **kwargs_websocket_bob)\n",
+    "#workers_virtual = [alice, bob]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#array_lines_proper_dimension = our data points(X)\n",
+    "#categories_numpy = our labels (Y)\n",
+    "langDataset =  LanguageDataset(array_lines_proper_dimension, categories_numpy)\n",
+    "\n",
+    "#assign the data points and the corresponding categories to workers.\n",
+    "federated_train_loader = sy.FederatedDataLoader(\n",
+    "            langDataset\n",
+    "            .federate(workers_virtual),\n",
+    "            batch_size=args.batch_size)  \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 4 - Training!\n",
```

Comment:
Can you use a single pattern for section titles? You started using <number> . <title>, in the end you're using <number> - <title>

Reply:
Done, made it all more coherent